engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -5618,35 +5631,76 @@ private void checkConcurrentJobsPerDatastoreThreshhold(final StoragePool destPoo
 
         final List<VmWorkJobVO> pendingWorkJobs = _workJobDao.listPendingWorkJobs(
                 VirtualMachine.Type.Instance, vm.getId(),
-                VmWorkAddVmToNetwork.class.getName());
+                VmWorkAddVmToNetwork.class.getName(), network.getUuid());
 
         VmWorkJobVO workJob = null;
         if (pendingWorkJobs != null && pendingWorkJobs.size() > 0) {
-            assert pendingWorkJobs.size() == 1;
-            workJob = pendingWorkJobs.get(0);
+            if (pendingWorkJobs.size() > 0) {
+                s_logger.warn(String.format("number of jobs to add net %s to vm %s are %d", network.getUuid(), vm.getInstanceName(), pendingWorkJobs.size()));
```

Comment:
```suggestion
                s_logger.warn(String.format("Number of pending jobs to add network %s to vm %s are %d", network.getName(), vm.getInstanceName(), pendingWorkJobs.size()));
```

Reply:
@suresh, `net.name` may be duplicate and less helpfull, don't you think? (rest of your sugestions I will apply when we agree)